cachito/workers/scm.py

```diff
@@ -131,7 +177,19 @@ def fetch_source(self):
             log.debug('The archive already exists at "%s"', self.archive_path)
             return
 
-        self.clone_and_archive()
+        # Find a previous archive created by a previous request
+        #
+        # The previous archive does not mean the one just before the request that
+        # schedules current task. The only reason for finding out such a file is
+        # to access the git history. So, anyone is ok.
+        filenames = glob.glob(os.path.join(self.package_dir, '*.tar.gz'))
+        filenames.append(None)
+        previous_archive = filenames[0]
```

Comment:
A nice optimization is to use the archive with the latest Git history.

```suggestion
        filenames = glob.glob(os.path.join(self.package_dir, '*.tar.gz')) or [None]
        # Get the newest archive since it's most likely to contain the newest Git history
        previous_archive = max(filenames, key=os.path.getctime)[0]
```